src/nvim/ex_getln.c

```diff
@@ -1022,9 +1022,12 @@ static int command_line_execute(VimState *state, int key)
 
   // <S-Tab> goes to last match, in a clumsy way
   if (s->c == K_S_TAB && KeyTyped) {
-    if (nextwild(&s->xpc, WILD_EXPAND_KEEP, 0, s->firstc != '@') == OK
-        && nextwild(&s->xpc, WILD_PREV, 0, s->firstc != '@') == OK
-        && nextwild(&s->xpc, WILD_PREV, 0, s->firstc != '@') == OK) {
+    if (nextwild(&s->xpc, WILD_EXPAND_KEEP, 0, s->firstc != '@') == OK) {
+      showmatches(&s->xpc, p_wmnu
+                  && ((wim_flags[s->wim_index] & WIM_LIST) == 0));
+      redrawcmd();
```

Comment:
Is this extra redraw needed? what happens otherwise?

Reply:
`<S-Tab>` seems to work without it but I decided to add it because the `<Tab>` path [also has one](https://github.com/neovim/neovim/blob/master/src/nvim/ex_getln.c#L990). It doesn't seem necessary either though (tests don't fail if I remove it). Should I amend my commit to remove both `redrawcmd()` calls?

Reply:
`wildoptions=pum` does not require any extra redraw compared to standand wildmenu, and this code seemingy worked before. The wildchar branch seems to be doing a lot of other stuff, I don't know if `redrawcmd()` can be elided there. But we shouldn't cargo-cult copy extra redraws unless we know they are needed in the new place also.